### PR TITLE
reader add readAndProcess method

### PR DESCRIPTION
**Pull request by yqylovy**

### CODE REVIEW

**Comment by troian** [comment]
Why no PR description??
What it changes, fixes?

**Comment by yqylovy** [comment]
hi,It is a code refactor pr, reduce duplicate code.

File: connection/connection.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 	"time"
 
 	"github.com/VolantMQ/vlapi/mqttp"
-	"github.com/VolantMQ/vlapi/plugin/auth"
+	vlauth "github.com/VolantMQ/vlapi/plugin/auth"
```
Comment (troian) [inline connection/connection.go:28]: What changes adding package prefix? It is already `vlauth`

**Comment by troian** [comment]
There are 2 patches not related to each other. `readAndProcess ` does not improve anything and I leave code as is for this moment until async socket operations available in go as before connection and after connection a bit 2 different ways.
Second patch `connection shold count qos1 package` looks like a fix. Can you make issue with description followed by PR with this patch.

**Comment by yqylovy** [comment]
OK，I will make a issue.

**Pull request closed without merging**